api/v2/api.go

```diff
@@ -146,19 +144,27 @@ func (api *API) getStatusHandler(params general_ops.GetStatusParams) middleware.
 		Config: &open_api_models.AlertmanagerConfig{
 			Original: &original,
 		},
-		Cluster: &open_api_models.ClusterStatus{
-			Name:   &name,
-			Status: &status,
-			Peers:  []*open_api_models.PeerStatus{},
-		},
 	}
 
-	for _, n := range api.peer.Peers() {
-		address := n.Address()
-		resp.Cluster.Peers = append(resp.Cluster.Peers, &open_api_models.PeerStatus{
-			Name:    &n.Name,
-			Address: &address,
-		})
+	// If api.peers == nil, Alertmanager cluster feature is disabled.
+	if api.peer != nil {
+		name := api.peer.Name()
+		status := api.peer.Status()
+
+		peers := []*open_api_models.PeerStatus{}
+		for _, n := range api.peer.Peers() {
+			address := n.Address()
+			peers = append(peers, &open_api_models.PeerStatus{
+				Name:    &n.Name,
+				Address: &address,
+			})
+		}
+
+		resp.Cluster = &open_api_models.ClusterStatus{
```

Comment:
The openapi spec (judging by the generated elm models) don't allow for `resp.Cluster` to be nil. Instead of modifying the spec, it might be more user-friendly to indicate that clustering is disabled.

Reply:
Could we just say
```go
if api.peers != nil {
    // stuff as usual
} else {
    cluster.status = "disabled"
    cluster.peers = []peers{}
    cluster.name = ...
}
```

Not sure what to do about `cluster.name`. Could either send an empty string and just hide the field, or also set it to "disabled".

I like acknowledging that there's no clustering in the ui, but I'm happy to discuss if you want to do what we previously did and hide the cluster status output entirely.

Reply:
Introducing a "disabled" cluster status sounds like a great idea to me.

What are your thoughts in regards to introducing a breaking change to the spec? In case that is fine for a bug fix, I would suggest making `cluster.peers` and `cluster.name` optional fields. If a breaking change is not an option I would be fine with an empty array and an empty ( or "disabled") string.

Reply:
I think we've been pretty clear that the openapi spec can (and probably will) change. I'm fine with a breaking change, we just have to remember to call it out and do a minor version bump :)

Reply:
> remember to call it out

Sure. I will do that in the changelog.

> do a minor version bump

That would be v0.17.0 then. Given that this is a bug fix and that we should get this out rather sooner than later I would vote for v0.16.1.

Reply:
>Given that this is a bug fix and that we should get this out rather sooner than later I would vote for v0.16.1.

If it's a breaking change then it shouldn't be patch level